### ISSUE TITLE
AuthError after migrating from Laravel 10 to 11 on Private Channel

### ISSUE THREAD

**Issue by jonjakoblich** [laravel/reverb#284]
### Reverb Version

1.4.4

### Laravel Version

11.35.1

### PHP Version

8.3.12

### Description

After having an existing Laravel 10 application, and following the [migration steps to upgrade to Laravel 11](https://laravel.com/docs/11.x/upgrade), I get the following error when listening to a private channel.

`Object { type: "AuthError", error: "JSON returned from channel-authorization endpoint was invalid, yet status code was 200. Data was: ", status: 200 }`

This private channel was previously working without error before upgrading.

This is occurring on my machine running Laravel Herd.
<img width="543" alt="Screenshot 2024-12-17 at 6 59 58 AM" src="https://github.com/user-attachments/assets/3b7e5698-767d-4d7e-abb6-78ebfa8e5854" />

### Steps To Reproduce

Demo repo: https://github.com/jonjakoblich/reverb-10

Steps taken in repository
1. Install fresh Laravel 10 application with latest Laravel 10 version
2. Install Jetstream w/ Inertia-Vue
3. Create an event to test
4. Modify welcome page to listen to event
5. Upgrade to Laravel 11

# To reproduce
1. Clone the repository linked above
3. Install and run the site using Laravel Herd (not tested in other environments)
4. Install composer packages, node packages, run migrations
5. Run `npm run build`
6. Register a user (& be logged in since the private channel is related to the user)
7. Navigate to welcome page (should be base URL of the newly installed app)
8. Open Tinker and run the test event `broadcast(new MyTestEvent(User::find(1)));`
9. Inspect in the console to see the error
10. Inspect the network tab to see that the web socket connection was not the expected response

**Comment by joedixon** [comment]
Hey @jonjakoblich, which URL is being requested when that error is returned?

**Comment by joedixon** [comment]
Thanks, but I meant is that request failing on the `broadcasting/auth` endpoint or connecting to the WebSocket server itself. Are you able to show the failing request in the network tab?

**Comment by jonjakoblich** [comment]
> Thanks, but I meant is that request failing on the `broadcasting/auth` endpoint or connecting to the WebSocket server itself. Are you able to show the failing request in the network tab?

@joedixon I'm attaching a few additional screenshots here of the broadcasting/auth endpoint. Hopefully this helps. Let me know if you need something else. 

<img width="547" alt="Screenshot 2024-12-19 at 3 45 51 PM" src="https://github.com/user-attachments/assets/0238a939-b242-4f60-b805-a47844748587" />
<img width="544" alt="Screenshot 2024-12-19 at 3 46 00 PM" src="https://github.com/user-attachments/assets/b3d47188-02b9-402f-bc73-d3f9fcb83410" />
<img width="541" alt="Screenshot 2024-12-19 at 3 46 09 PM" src="https://github.com/user-attachments/assets/49e94d44-d06c-4304-8621-54a680d266cc" />

**Comment by jonjakoblich** [comment]
> I'm sorry, it's still not clear to me where the error is being generated. Are any of the network requests returning failed status codes?

@joedixon No, they are not. The blue highlighted request in each of those screenshots is the request. You can see the 101 status in the first one (the web socket connection I believe) and the 200 status in the successive ones on each (the /broadcasting/auth endpoint). The screenshots are from Firefox Developer Edition. Beyond that, I've not been able to trace the error. It looks like it connects to the /broadcasting/auth endpoint, gets a 200 response, but not the anticipated data back.

I don't know if you've had a chance to clone and install the demo repo I provided, but that does illustrate the issue when using Laravel Herd. I have not tried other setups.

**Comment by i6media** [comment]
I was facing the same problem, make sure you have `BROADCAST_CONNECTION=reverb` in you `.env`

**Comment by jonjakoblich** [comment]
> I was facing the same problem, make sure you have `BROADCAST_CONNECTION=reverb` in your `.env`

I already have it set up that way. Thank you.

**Comment by joedixon** [comment]
Just had a chance to pull down your example repo and looks like you need to set the following variable or update the variable name in your `broadcasting.php` config file:

```
BROADCAST_DRIVER=reverb
```

**Comment by francoism90** [comment]
@joedixon I just spent 3 hours on Reverb/Echo not working, and it turned out I didn't set `BROADCAST_DRIVER=reverb`.. ugh.
